packages/palette_generator/lib/palette_generator.dart

```diff
@@ -1034,16 +1016,16 @@ class _ColorCount {
 }
 
 class _ColorHistogram {
-  final Map<int, Map<int, Map<int, _ColorCount>>> _hist =
-      <int, Map<int, Map<int, _ColorCount>>>{};
+  final Map<int, Map<int, Map<int, _ColorCount?>>> _hist =
```

Comment:
Why would there be null counts in a histogram?

Reply:
The value _ColorCount is set as null in https://github.com/flutter/packages/blob/48a2148811a09b6354e8b02551e34d156ed56916/packages/palette_generator/lib/palette_generator.dart#L1085 and which is used for ignoring some colors seems like
https://github.com/flutter/packages/blob/48a2148811a09b6354e8b02551e34d156ed56916/packages/palette_generator/lib/palette_generator.dart#L1218

Reply:
Given that it's called "remove", it seems like the intent would be to remove the entries. Is there anything that actually treats null values differently from missing values? If not, this was probably just (arguably incorrect) shorthand for removal.

Reply:
Please check the recent https://github.com/flutter/packages/pull/287/commits/bbb054711afe3e3011d8ef4b0414bc5becdd8521 commit if it is a correct implementation

Reply:
`_hist[key.red]?[key.blue]?.remove(key.green)`

Fixing that removal should allow keys to be non-nullable.